SuccessEycaActivationActivity/eyca.ts

```diff
@@ -0,0 +1,145 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { format } from "date-fns";
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromEither,
+  fromLeft,
+  TaskEither,
+  taskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { RedisClient } from "redis";
+import { EycaAPIClient } from "../clients/eyca";
+import { Timestamp } from "../generated/definitions/Timestamp";
+import { CcdbNumber } from "../generated/eyca-api/CcdbNumber";
+import { ErrorResponse } from "../generated/eyca-api/ErrorResponse";
+import { errorsToError } from "../utils/conversions";
+import { getTask, setWithExpirationTask } from "../utils/redis_storage";
+
+export const CCDB_SESSION_ID_KEY = "CCDB_SESSION_ID";
+export const CCDB_SESSION_ID_TTL = 1700;
+
+/**
+ * Performs a login through EYCA CCDB Login API
+ * via username and password credentials.
+ * A success response includes a session_id token
+ */
+const ccdbLogin = (
+  eycaClient: ReturnType<EycaAPIClient>,
+  username: NonEmptyString,
+  password: NonEmptyString
+): TaskEither<Error, NonEmptyString> =>
+  tryCatch(
+    () =>
+      eycaClient.authLogin({
+        password,
+        username
+      }),
+    toError
+  )
+    .mapLeft(err => new Error(`Cannot call EYCA authLogin API ${err.message}`))
+    .chain(_ => fromEither(_).mapLeft(errorsToError))
+    .chain<NonEmptyString>(res =>
+      res.status !== 200 || ErrorResponse.is(res.value.api_response)
+        ? fromLeft(
+            new Error(
+              `Error on EYCA authLogin API|STATUS=${res.status}, DETAIL=${res.value.api_response.text}`
+            )
+          )
+        : taskEither.of(res.value.api_response.text)
+    );
+
+/**
+ * Retrieves a previous stored session_id on Redis cache.
+ * If missing a new login is performed and the related session_id
+ * is stored on Redis.
+ */
+const retrieveCcdbSessionId = (
+  redisClient: RedisClient,
+  eycaClient: ReturnType<EycaAPIClient>,
+  username: NonEmptyString,
+  password: NonEmptyString
+): TaskEither<Error, NonEmptyString> =>
+  getTask(redisClient, CCDB_SESSION_ID_KEY).chain(maybeSessionId =>
+    maybeSessionId.foldL(
+      () =>
+        ccdbLogin(eycaClient, username, password).chain(_ =>
+          setWithExpirationTask(
+            redisClient,
+            CCDB_SESSION_ID_KEY,
+            _,
+            CCDB_SESSION_ID_TTL
+          ).map(() => _)
```

Comment:
Only a small readability improvement
```suggestion
        ccdbLogin(eycaClient, username, password).chain(sessionId =>
          setWithExpirationTask(
            redisClient,
            CCDB_SESSION_ID_KEY,
            sessionId,
            CCDB_SESSION_ID_TTL
          ).map(() => sessionId)
```